howto-gallery/scripts/BlockEvent.ts

```diff
@@ -1,19 +1,21 @@
 import * as mc from "@minecraft/server";
 import { MinecraftBlockTypes, MinecraftDimensionTypes } from "@minecraft/vanilla-data";
 
-const overworld = mc.world.getDimension(MinecraftDimensionTypes.Overworld);
-
 /**
  * A simple button push before event
  * @param {(message: string, status?: number) => void} log: Logger function. If status is positive, test is a success. If status is negative, test is a failure.
- * @param {mc.Location} location Location to center this sample code around.
+ * @param {mc.DimensionLocation} targetLocation Location to center this sample code around.
  * @see https://learn.microsoft.com/minecraft/creator/scriptapi/minecraft/server/buttonpushaftereventsignal
  * @see https://learn.microsoft.com/minecraft/creator/scriptapi/minecraft/server/buttonpushafterevent
  */
-export function buttonPushEvent(log: (message: string, status?: number) => void, targetLocation: mc.Vector3) {
+export function buttonPushEvent(log: (message: string, status?: number) => void, targetLocation: mc.DimensionLocation) {
   // set up a button on cobblestone
-  let cobblestone = overworld.getBlock(targetLocation);
-  let button = overworld.getBlock({ x: targetLocation.x, y: targetLocation.y + 1, z: targetLocation.z });
+  const cobblestone = targetLocation.dimension.getBlock(targetLocation);
+  const button = targetLocation.dimension.getBlock({
+    x: targetLocation.x,
+    y: targetLocation.y + 1,
+    z: targetLocation.z,
+  });
```

Comment:
Here we can use `Vector3Utils.add` as sample

Reply:
For basic operations like this I'd say it's probably not worth the dependency